src/Education/XamlGenerator/IRuleHelpXamlBuilder.cs

```diff
@@ -0,0 +1,472 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Windows.Documents;
+using System.Windows.Markup;
+using System.Xml;
+using SonarLint.VisualStudio.Rules;
+
+namespace SonarLint.VisualStudio.Education.XamlGenerator
+{
+    public interface IRuleHelpXamlBuilder
+    {
+        /// <summary>
+        /// Generates a XAML document containing the help information for the specified rule
+        /// </summary>
+        /// <remarks>Assumes that the <see cref="IRuleHelp.HtmlDescription"/> is parseable as XML</remarks>
+        FlowDocument Create(IRuleHelp ruleHelp);
+    }
+
+    internal partial class RuleHelpXamlBuilder : IRuleHelpXamlBuilder
```

Comment:
I'll clean up and simplify the implementation once I've made the changes to support styling and theming.

At a high-level, the idea is fairly simple:
* read through the input HTML doc (treating it as XML)
* for each HTML element, write out at the appropriate XAML element(s) and possibly attributes.

Often there is a 1-1 mapping e.g.
* `<h2>` -> `<Paragraph FontSize="20pt">`
* `<ul` -> `<List>`
* `<br>` -> `<LineBreak>`

Lists and tables are a bit more complicated, but not much.

The builder only handles a subset of HTML tags, but it will throw for any it does not recognise.